lib/AST/Type.cpp

```diff
@@ -5158,7 +5199,8 @@ case TypeKind::Id:
     // First, substitute down the pattern type to gather the mapping from
     // contained substitutable types to packs.
     auto gather = ExpansionGatherer{fn};
-    Type transformedPat = expand->getPatternType().transformRec(gather);
+    Type transformedPat =
+        expand->getPatternType().transformWithPosition(pos, gather);
```

Comment:
I am intuitively treating packs as invariant (because they seem like they cannot within another type), and expansions as covariant here; let me know if anything sounds fishy about that.